sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/socket.scala

```diff
@@ -126,16 +128,17 @@ class TextSocketSource(host: String, port: Int, includeTimestamp: Boolean, sqlCo
       batches.slice(sliceStart, sliceEnd)
     }
 
-    import sqlContext.implicits._
-    val rawBatch = sqlContext.createDataset(rawList)
+    val rdd = sqlContext.sparkContext.parallelize(rawList).map(
+        v => InternalRow(UTF8String.fromString(v._1), v._2.getTime()))
+    val rawBatch = sqlContext.internalCreateDataFrame(rdd, schema, isStreaming = true)
 
     // Underlying MemoryStream has schema (String, Timestamp); strip out the timestamp
     // if requested.
     if (includeTimestamp) {
       rawBatch.toDF("value", "timestamp")
     } else {
       // Strip out timestamp
-      rawBatch.select("_1").toDF("value")
+      rawBatch.select("value").toDF()
```

Comment:
`toDF` is unnecessary since it's already a DataFrame.

Reply:
Done.